internal/output/progress_with_status_bars.go

```diff
@@ -0,0 +1,22 @@
+package output
+
+type ProgressWithStatusBars interface {
+	Progress
+
+	StatusBarUpdatef(i int, format string, args ...interface{})
+	StatusBarCompletef(i int, format string, args ...interface{})
+	StatusBarResetf(i int, label, format string, args ...interface{})
+}
+
+func newProgressWithStatusBars(bars []ProgressBar, statusBars []*StatusBar, o *Output, opts *ProgressOpts) ProgressWithStatusBars {
+	barPtrs := make([]*ProgressBar, len(bars))
+	for i := range bars {
+		barPtrs[i] = &bars[i]
+	}
+
+	if !o.caps.Isatty {
+		newProgressWithStatusBarsSimple(barPtrs, statusBars, o)
+	}
+
+	return newProgressWithStatusBarsTTY(barPtrs, statusBars, o, opts)
```

Comment:
Note to reviewers: `progressWithStatusbarsTTY` and `progressWithSatusBarsSimple` have a lot of similarities with `progressTTY` and `progressSimple` respectively. I've tried to extract some of the common stuff and embed `progressTTY` in `progressWithStatusBarsTTY`. That helped but not as much as it would've in a language where inheritance was a first class concept.

The problem in Go is this:

```go
type inner struct {
	name string
}

func (i *inner) String() string { return i.format() }
func (i *inner) format() string { return fmt.Sprintf("my name is %s", i.name) }

type outer struct {
	*inner
}

func (o *outer) format() string { return "i am outer!" }

func main() {
	i := &inner{name: "inner"}
	fmt.Println(i)

	o := &outer{inner: i}
	fmt.Println(o)
}
```

That prints:

```
my name is inner
my name is inner
```

[See this StackOverflow answer for details](https://stackoverflow.com/a/29390981). Short version: Go doesn't have virtual methods.

I might try to "invert" the code and have the structs' methods call functions, passing themselves in, but I'm not sure it's worth it.

Because right now I don't think the duplication is a problem. The code that is duplicated depends on the callee's state and doesn't contain a lot of duplicated logic. The hairy bits are not duplicated.